curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo/64e4ecd7735a566c9266a338.md

```diff
@@ -0,0 +1,326 @@
+---
+id: 64e4ecd7735a566c9266a338
+title: Step 10
+challengeType: 0
+dashedName: step-10
+---
+
+# --description--
+
+Now that you've worked on opening and closing the modal, it's time to get the values from the input fields, save them into the `taskData` array, and display them on the page.
+
+To start, add a `submit` event listener to your `taskForm` element and pass in `e` as the parameter of your arrow function. Inside the curly braces, use the `preventDefault()` method to stop the browser from refreshing the page after submitting the form.
+
+# --hints--
+
+You should call the `addEventListener()` method on your `taskForm` variable.
+
+```js
+assert.match(code, /taskForm\.addEventListener\(/)
+```
+
+Your event listener should listen for a `submit` event.
+
+```js
+assert.match(code, /taskForm\.addEventListener\(('|"|`)submit\1/)
+```
+
+You should use arrow syntax to set your event listener to an empty pair of curly braces with `e` as the parameter.
+
+```js
+assert.match(code, /taskForm\.addEventListener\(('|"|`)submit\1,\s*\(e\)\s*=>\s*\{/)
+```
+
+You should use the `e.preventDefault()` method to stop the browser from reloading the page.
+
+```js
+assert.match(code, /taskForm\.addEventListener\(('|"|`)submit\1,\s*\(e\)\s*=>\s*\{\s*e\.preventDefault\(\)\s*\}\);?/)
```

Comment:
When you run the `FCC_BLOCK='Learn localStorage by Building a Todo App' pnpm run test:curriculum` command it throws an error for this test because the test does not match up with step 11 and onwards solutions.

<img width="775" alt="Screenshot 2023-09-28 at 10 01 31 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/4be07028-3581-4373-a008-a1c334a68bc8">

To fix it you should add the option use of the semicolon after the `e.preventDefault()`

```suggestion
assert.match(code, /taskForm\.addEventListener\(('|"|`)submit\1,\s*\(e\)\s*=>\s*\{\s*e\.preventDefault\(\);?\s*\}\);?/)
```